src/earn/EarnEnterAmount.test.tsx

```diff
@@ -372,9 +387,108 @@ describe('EarnEnterAmount', () => {
+      const { getByTestId, getByText } = render(
+        <Provider store={store}>
+          <MockedNavigator component={EarnEnterAmount} params={withdrawParams} />
+        </Provider>
+      )
+
+      fireEvent.changeText(getByTestId('EarnEnterAmount/TokenAmountInput'), '8')
+
+      await waitFor(() => expect(getByText('earnFlow.enterAmount.continue')).not.toBeDisabled())
+
+      expect(getByTestId('EarnEnterAmount/Withdraw/Crypto')).toBeTruthy()
+      expect(getByTestId('EarnEnterAmount/Withdraw/Crypto')).toHaveTextContent('11.00 USDC')
+
+      expect(getByTestId('EarnEnterAmount/Withdraw/Fiat')).toBeTruthy()
+      expect(getByTestId('EarnEnterAmount/Withdraw/Fiat')).toBeTruthy()
+      expect(getByTestId('EarnEnterAmount/Withdraw/Fiat')).toHaveTextContent('₱14.63')
+
+      expect(getByTestId('EarnEnterAmount/Fees')).toBeTruthy()
+      expect(getByTestId('EarnEnterAmount/Fees')).toHaveTextContent('₱0.012')
+
+      fireEvent.press(getByText('earnFlow.enterAmount.continue'))
+
+      await waitFor(() => expect(AppAnalytics.track).toHaveBeenCalledTimes(1))
+      expect(AppAnalytics.track).toHaveBeenCalledWith(EarnEvents.earn_enter_amount_continue_press, {
+        amountEnteredIn: 'token',
+        amountInUsd: '8.00',
+        networkId: NetworkId['arbitrum-sepolia'],
+        depositTokenId: mockArbUsdcTokenId,
+        providerId: mockEarnPositions[0].appId,
+        poolId: mockEarnPositions[0].positionId,
+        fromTokenId: 'arbitrum-sepolia:0x94a9d9ac8a22534e3faca9f4e7f2e2cf85d5e4c8',
+        fromTokenAmount: '8',
+        mode: 'withdraw',
+      })
+
+      //TODO(ACT-1389): check navigation to withdrawal confirmation screen
+    })
+
+    it('should allow the user to set an input value over the pool balance if pricePerShare is great than 1', async () => {
+      const { getByTestId, queryByTestId } = render(
+        <Provider store={store}>
+          <MockedNavigator
+            component={EarnEnterAmount}
+            params={{ pool: mockPoolWithHighPricePerShare, mode: 'withdraw' }}
+          />
+        </Provider>
+      )
+
+      fireEvent.changeText(getByTestId('EarnEnterAmount/TokenAmountInput'), '15')
+      expect(queryByTestId('EarnEnterAmount/NotEnoughBalanceWarning')).toBeFalsy()
+      expect(getByTestId('EarnEnterAmount/Continue')).toBeEnabled()
+    })
+
+    it('should not allow the user to set an input amount higher than pool balance * pricePerShare', async () => {
+      const { getByTestId } = render(
+        <Provider store={store}>
+          <MockedNavigator
+            component={EarnEnterAmount}
+            params={{ pool: mockPoolWithHighPricePerShare, mode: 'withdraw' }}
+          />
+        </Provider>
+      )
+
+      fireEvent.changeText(getByTestId('EarnEnterAmount/TokenAmountInput'), '20.001')
+      expect(getByTestId('EarnEnterAmount/NotEnoughBalanceWarning')).toBeTruthy()
+      expect(getByTestId('EarnEnterAmount/Continue')).toBeDisabled()
+    })
```

Comment:
can we add tests for checking transaction preparation is called with the right amounts?